packages/@o3r/extractors/src/core/comparator/metadata-comparison.helper.ts

```diff
@@ -134,13 +135,23 @@ Detection of package manager runner will fallback on the one used to execute the
   context.logger.info(`Fetching ${packageLocator} from the registry.`);
   const packageJsonFileName = 'package.json';
   const previousFile = await getFilesFromRegistry(packageLocator, [options.metadataPath, packageJsonFileName], packageManager, context.workspaceRoot);
-
-  context.logger.info(`Resolved metadata from version ${JSON.parse(previousFile[packageJsonFileName]).version}.`);
+  const previousPackageJson = JSON.parse(previousFile[packageJsonFileName]) as PackageJson & {cmsMetadata: CmsMetadataData};
+  context.logger.info(`Successfully fetched ${packageLocator} with version ${previousPackageJson.version}.`);
+
+  let metadata: MetadataFile;
+  const previousMetadataEntryFromPackageJson = previousPackageJson.cmsMetadata?.[options.packageJsonEntry];
+  if (previousMetadataEntryFromPackageJson) {
+    const previousFileFromPackageJson = await getFilesFromRegistry(packageLocator, [previousMetadataEntryFromPackageJson], packageManager, context.workspaceRoot);
+    metadata = JSON.parse(previousFileFromPackageJson[previousMetadataEntryFromPackageJson]) as MetadataFile;
+    context.logger.info(`Resolved metadata from "${previousPackageJson.cmsMetadata[options.packageJsonEntry]}".`);
+  } else {
```

Comment:
Is it necessary to have the filed in package.json to have higher priority than the option passed in command line?. Usually we have the option in command line with the higher priority.